spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindingService.java

```diff
@@ -152,6 +160,47 @@ public BindingService(
 		});
 	}
 
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	public <T> Binding<T> doBindPollableConsumer(T input, String inputName, Binder<T, ConsumerProperties, ?> binder,
```

Comment:
Do we really need to spell it out as `doBindPollableConsumer`? The code up above `if (input instanceof PollableSource) {` would be much simpler if it was simply `doBindConsumer`. Since it's `PollableSource` it is rather obvious what type of consumer it will be, right?

Reply:
We already have a `doBindConsumer` for the message-driven consumer; all this generic stuff makes it hard to collapse them into one method.